assets/js/src/announcements/feature_announcement.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import classNames from 'classnames';
+import MailPoet from 'mailpoet';
+import PropTypes from 'prop-types';
+
+class FeatureAnnouncement extends React.Component {
+  constructor(props) {
+    super(props);
+    this.loadBeamer = this.loadBeamer.bind(this);
+    this.beamerCallback = this.beamerCallback.bind(this);
+
+    this.state = {
+      showDot: props.hasNews,
+      beamerLoaded: typeof window.Beamer !== 'undefined',
+    };
+  }
+
+  loadBeamer(e) {
+    e.preventDefault();
+    if (!this.state.beamerLoaded) {
+      window.beamer_config = {
+        product_id: 'VvHbhYWy7118',
+        selector: 'beamer-selector',
+        language: window.mailpoet_user_locale,
+        callback: this.beamerCallback,
+      };
+      MailPoet.Modal.loading(true);
+      this.setState({ beamerLoaded: true });
+      window.mailpoet_feature_announcement_has_news = false;
+      const s = document.createElement('script');
+      s.type = 'text/javascript';
+      s.src = 'https://app.getbeamer.com/js/beamer-embed.js';
+      document.getElementsByTagName('body')[0].appendChild(s);
+    }
+  }
+
+  beamerCallback() {
+    this.setState({ showDot: false });
+    MailPoet.Modal.loading(false);
+    window.Beamer.show();
+    const data = { last_announcement_seen: {} };
```

Comment:
I've found an issue here. It seems that settings API just [overwrite the value stored on a first level key](https://github.com/mailpoet/mailpoet/blob/1f19a851c31b1dafc386a3d5c8d8de829aee419e/lib/API/JSON/v1/Settings.php#L31).  So here it always overwrites `last_announcement_seen` which results in a behaviour that when one user sees the announcement panel he will overwrite timestamp for all other users.

So I think we need to load the previously stored `last_announcement_seen` add or update record for the user and then save it. It seems that api currently support only fetching all...

Another option could be adding user id as a suffix to a settings key.

Reply:
Nice catch! The settings are already loaded on this page in the `mailpoet_settings` global variable so we can take the necessary data from there.

Reply:
In another PR (not yet merged) I removed that global variable, so it might not work after merging https://github.com/mailpoet/mailpoet/pull/1648/files#diff-ce4dc7e46f5c02070525fe5b485fdfdc

Reply:
@NeosinneR Thanks for letting me know! I'll pass announcement data in another variable then.

Reply:
@NeosinneR Oh I missed that one in your PR. I thought you removed It only from layout.html. I think it is better to remove such a big object, but I don’t know what is it needed for on the newsletters page. It was originally there and I’m a bit afraid that removing it could break some things

Reply:
Ok then. Let’s keep it removed.

Reply:
Fixed.